Geometry/VeryForwardGeometry/interface/CTPPSPixelSimTopology.h

```diff
@@ -0,0 +1,214 @@
+#ifndef Geometry_VeryForwardGeometry_CTPPSPixelSimTopology_h
+#define Geometry_VeryForwardGeometry_CTPPSPixelSimTopology_h
+
+#include "TMath.h"
+#include "Geometry/VeryForwardGeometry/interface/CTPPSPixelTopology.h"
+
+class CTPPSPixelSimTopology : public CTPPSPixelTopology
+{
+  public:
+    /* simX and simY are the coordinates as in the simulation:
+        _________
+       |         |
+       |         |  y
+       |_________|
+
+       x
+    */
+    class PixelInfo
+    {
+      public:
+        PixelInfo( double lower_simX_border, double higher_simX_border, double lower_simY_border, double higher_simY_border, double eff_factor, unsigned short pixel_row_no, unsigned short pixel_col_no ) :
+          lower_simX_border_( lower_simX_border ), higher_simX_border_( higher_simX_border ),
+          lower_simY_border_( lower_simY_border ), higher_simY_border_( higher_simY_border ),
+          eff_factor_( eff_factor ),
+          pixel_row_no_( pixel_row_no ), pixel_col_no_( pixel_col_no ),
+          pixel_index_( pixel_col_no*CTPPSPixelTopology::no_of_pixels_simX_+pixel_row_no )
+        {}
+
+        inline double higherSimXBorder() const { return higher_simX_border_; }
+        inline double lowerSimXBorder() const { return lower_simX_border_; }
+        inline double higherSimYBorder() const { return higher_simY_border_; }
+        inline double lowerSimYBorder() const { return lower_simY_border_; }
+        inline double effFactor() const { return eff_factor_; }
+        inline unsigned short pixelRowNo() const { return pixel_row_no_; }
+        inline unsigned short pixelColNo() const { return pixel_col_no_; }
+        inline unsigned short pixelIndex() const { return pixel_index_; }
+
+      private:
+        double lower_simX_border_;
+        double higher_simX_border_;
+        double lower_simY_border_;
+        double higher_simY_border_;
+        double eff_factor_;
+        unsigned short pixel_row_no_;
+        unsigned short pixel_col_no_;
+        unsigned short pixel_index_;
+    };
+
+  public:
+    CTPPSPixelSimTopology();
+    ~CTPPSPixelSimTopology() {}
+
+    std::vector<PixelInfo> getPixelsInvolved( double x, double y, double sigma, double& hit_pos_x, double& hit_pos_y );
```

Comment:
Hi @davidlange6!
Sorry for my insistence... Since our pixels clusterisation strongly depends on this PR we were a bit tight in time.
This method will be used later in our digitisation module (not yet deployed). Its implementation might slightly change in the future, but we currently use this one as a placeholder (in fact, the `double sigma` parameter is not even used yet).